### ISSUE TITLE
TB68 + TbSync + autoadmin: Does not work after restart.

### ISSUE THREAD

**Issue by ohmer1** [jobisoft/TbSync#343]
I'm using Ubuntu 18.04. They upgraded to Thunderbird 68.2.1 from 60.x a few days ago. After a second startup of Thunderbird, TbSync is not working anymore. I don't see it on the Thunderbird's status bar and when clicking on menu Tools -> Modules settings (translated from "Préférences des modules") -> TbSync, nothing happen except this exception I can find in the error console:

```
TypeError: TbSync.manager is undefined accountManager.js:21:5
    onunloadoptions chrome://tbsync/content/manager/accountManager.js:21
    onunload chrome://tbsync/content/manager/addonoptions.xul:1
```

If I close Thunderbird and delete the directory `~/.cache/thunderbird/<profile name>/startupCache`, TbSync works until I close Thunderbird. I then have to drop that directory again.

**Comment by jobisoft** [comment]
are you using tbsync (and its providers) from your distro repository or the official versions from atn?

Which versions?

**Comment by jobisoft** [comment]
ATN= addons.thunderbird.net

**Comment by ohmer1** [comment]
From addons.thunderbird.net. 

TbSync: version 2.8
CalDav provider: version 2.6

**Comment by jobisoft** [comment]
I have no idea. Are there any other errors? Can you enable debug logging (tbsync account manager -> help) and check if any thing useful is logged into the debug.log file in the tbsync folder in your thunderbird profile?

**Comment by ohmer1** [comment]
Debug logs are written only when tbsync is working. If I don't drop the problematic folder before starting Thunderbird, tbsync doesn't work and nothing is written in the debug logs.

**Comment by jobisoft** [comment]
Is there anything useful written while closing TB? Which could cause the next start to fail?

**Comment by ohmer1** [comment]
Nothing useful I think:
```
** Wed Dec 04 2019 14:39:50 GMT-0500 (heure normale de lEst nord-am�ricain) **
[TbSync init] : Start (2.8)

** Wed Dec 04 2019 14:39:50 GMT-0500 (heure normale de lEst nord-am�ricain) **
[Thunderbird] : 68.2.1 on Linux

** Wed Dec 04 2019 14:39:50 GMT-0500 (heure normale de lEst nord-am�ricain) **
[Loading module <db>] : OK

** Wed Dec 04 2019 14:39:50 GMT-0500 (heure normale de lEst nord-am�ricain) **
[Loading module <abAutoComplete>] : OK

** Wed Dec 04 2019 14:39:50 GMT-0500 (heure normale de lEst nord-am�ricain) **
[Loading module <addressbook>] : OK

** Wed Dec 04 2019 14:39:50 GMT-0500 (heure normale de lEst nord-am�ricain) **
[Loading module <lightning>] : OK

** Wed Dec 04 2019 14:39:50 GMT-0500 (heure normale de lEst nord-am�ricain) **
[Loading module <cardbook>] : OK

** Wed Dec 04 2019 14:39:50 GMT-0500 (heure normale de lEst nord-am�ricain) **
[Loading module <eventlog>] : OK

** Wed Dec 04 2019 14:39:50 GMT-0500 (heure normale de lEst nord-am�ricain) **
[Loading module <core>] : OK

** Wed Dec 04 2019 14:39:50 GMT-0500 (heure normale de lEst nord-am�ricain) **
[Loading module <passwordManager>] : OK

** Wed Dec 04 2019 14:39:50 GMT-0500 (heure normale de lEst nord-am�ricain) **
[Loading module <network>] : OK

** Wed Dec 04 2019 14:39:50 GMT-0500 (heure normale de lEst nord-am�ricain) **
[Loading module <tools>] : OK

** Wed Dec 04 2019 14:39:50 GMT-0500 (heure normale de lEst nord-am�ricain) **
[Loading module <manager>] : OK

** Wed Dec 04 2019 14:39:50 GMT-0500 (heure normale de lEst nord-am�ricain) **
[Loading module <providers>] : OK

** Wed Dec 04 2019 14:39:50 GMT-0500 (heure normale de lEst nord-am�ricain) **
[Loading module <messenger>] : OK

** Wed Dec 04 2019 14:39:50 GMT-0500 (heure normale de lEst nord-am�ricain) **
[Initializing module] : <db>

** Wed Dec 04 2019 14:39:50 GMT-0500 (heure normale de lEst nord-am�ricain) **
[Initializing module] : <abAutoComplete>

** Wed Dec 04 2019 14:39:50 GMT-0500 (heure normale de lEst nord-am�ricain) **
[Initializing module] : <addressbook>

** Wed Dec 04 2019 14:39:50 GMT-0500 (heure normale de lEst nord-am�ricain) **
[Initializing module] : <lightning>

** Wed Dec 04 2019 14:39:50 GMT-0500 (heure normale de lEst nord-am�ricain) **
[Check4Lightning] : 68.2.1

** Wed Dec 04 2019 14:39:50 GMT-0500 (heure normale de lEst nord-am�ricain) **
[Check4Lightning] : Done

** Wed Dec 04 2019 14:39:50 GMT-0500 (heure normale de lEst nord-am�ricain) **
[Initializing module] : <cardbook>

** Wed Dec 04 2019 14:39:50 GMT-0500 (heure normale de lEst nord-am�ricain) **
[Initializing module] : <eventlog>

** Wed Dec 04 2019 14:39:50 GMT-0500 (heure normale de lEst nord-am�ricain) **
[Initializing module] : <core>

** Wed Dec 04 2019 14:39:50 GMT-0500 (heure normale de lEst nord-am�ricain) **
[Initializing module] : <passwordManager>

** Wed Dec 04 2019 14:39:50 GMT-0500 (heure normale de lEst nord-am�ricain) **
[Initializing module] : <network>

** Wed Dec 04 2019 14:39:50 GMT-0500 (heure normale de lEst nord-am�ricain) **
[Initializing module] : <tools>

** Wed Dec 04 2019 14:39:50 GMT-0500 (heure normale de lEst nord-am�ricain) **
[Initializing module] : <manager>

** Wed Dec 04 2019 14:39:50 GMT-0500 (heure normale de lEst nord-am�ricain) **
[Initializing module] : <providers>

** Wed Dec 04 2019 14:39:50 GMT-0500 (heure normale de lEst nord-am�ricain) **
[Initializing module] : <messenger>

** Wed Dec 04 2019 14:39:50 GMT-0500 (heure normale de lEst nord-am�ricain) **
[TbSync init] : Done

** Wed Dec 04 2019 14:39:50 GMT-0500 (heure normale de lEst nord-am�ricain) **
[Loaded provider] : dav::CalDAV & CardDAV (1.6)

############# Thunderbird is closed here ##############

** Wed Dec 04 2019 14:40:11 GMT-0500 (heure normale de lEst nord-am�ricain) **
[TbSync shutdown] : Unloading TbSync modules.

** Wed Dec 04 2019 14:40:11 GMT-0500 (heure normale de lEst nord-am�ricain) **
[Unloading module <messenger>] : OK

** Wed Dec 04 2019 14:40:11 GMT-0500 (heure normale de lEst nord-am�ricain) **
[Unloading provider] : dav

** Wed Dec 04 2019 14:40:11 GMT-0500 (GMT-05:00) **
[Unloading module <providers>] : OK

** Wed Dec 04 2019 14:40:11 GMT-0500 (GMT-05:00) **
[Unloading module <manager>] : OK

** Wed Dec 04 2019 14:40:11 GMT-0500 (GMT-05:00) **
[Unloading module <tools>] : OK

** Wed Dec 04 2019 14:40:11 GMT-0500 (GMT-05:00) **
[Unloading module <network>] : OK

** Wed Dec 04 2019 14:40:11 GMT-0500 (GMT-05:00) **
[Unloading module <passwordManager>] : OK

** Wed Dec 04 2019 14:40:11 GMT-0500 (GMT-05:00) **
[Unloading module <core>] : OK

** Wed Dec 04 2019 14:40:11 GMT-0500 (GMT-05:00) **
[Unloading module <eventlog>] : OK

** Wed Dec 04 2019 14:40:11 GMT-0500 (GMT-05:00) **
[Unloading module <cardbook>] : OK

** Wed Dec 04 2019 14:40:11 GMT-0500 (GMT-05:00) **
[Unloading module <lightning>] : OK

** Wed Dec 04 2019 14:40:11 GMT-0500 (GMT-05:00) **
[Unloading module <addressbook>] : OK

** Wed Dec 04 2019 14:40:11 GMT-0500 (GMT-05:00) **
[Unloading module <abAutoComplete>] : OK

** Wed Dec 04 2019 14:40:11 GMT-0500 (GMT-05:00) **
[Unloading module <db>] : OK
```

**Comment by jobisoft** [comment]
no, nothing useful. Can you try to create a new fresh profile and install only tbsync + provider?

**Comment by ohmer1** [comment]
Tried with a new profile without using autoadmin. It works without problems.

But we are using the autoadmin feature of Thunderbird to autoconfigure the profile for all users. Seem like the bug occurs only when using automadmin (even from new profile). I created a sample autoadmin configuration with a minimal settings and this still bug. I don't know what's wrong.

/usr/lib/thunderbird/defaults/pref/local-settings.js:
```
// Load config file.
pref("general.config.obscure_value", 0);
pref("general.config.filename", "mozilla.cfg");
```

/usr/lib/thunderbird/mozilla.cfg:
```
// Autoconfig
var env_user = '';
if(getenv("USER") != "")
{
  env_user = getenv("USER");
}
else
{
  env_user = getenv("USERNAME");
}

var mail = '';
if (env_user!='')
{
  mail =  env_user + '@domain.tld';
}


lockPref("mail.identity.useremail", mail);
lockPref("autoadmin.append_emailaddr", true);
lockPref("autoadmin.global_config_url", "https://mozilla-autoconfig-staging.domain.tld/test.js");
lockPref("autoadmin.failover_to_cached", true);
lockPref("autoadmin.offline_failover", true);
```

The test.js file content:
```
// Local Account
lockPref("mail.account.account1.server", "server1");
lockPref("mail.accountmanager.localfoldersserver", "server1");
lockPref("mail.server.server1.directory-rel", "[ProfD]Mail/Local Folders");
lockPref("mail.server.server1.hostname", "Local Folders");
lockPref("mail.server.server1.name", "Dossiers locaux");
lockPref("mail.server.server1.type", "none");
lockPref("mail.server.server1.userName", "nobody");

// SMTP config
lockPref("mail.smtp.defaultserver", "smtp1");
lockPref("mail.smtpservers", "smtp1");
lockPref("mail.smtpserver.smtp1.try_ssl", 3);
lockPref("mail.smtpserver.smtp1.authMethod", 3);
lockPref("mail.smtpserver.smtp1.port", 465);
lockPref("mail.smtpserver.smtp1.hostname", "mail.domain.tld");
lockPref("mail.smtpserver.smtp1.username", "username@domain.tld");

// Default account is acocunt2 (the personal account)
lockPref("mail.accountmanager.defaultaccount", "account2");

// Personnal account
lockPref("mail.account.account2.server", "server2");
lockPref("mail.server.server2.hostname", "mail.domain.tld");
lockPref("mail.server.server2.name", "username@domain.tld");
lockPref("mail.server.server2.port", 993);
lockPref("mail.server.server2.socketType", 3);
lockPref("mail.server.server2.authMethod", 3);
lockPref("mail.server.server2.type", "imap");
lockPref("mail.server.server2.userName", "username@domain.tld");

lockPref("mail.identity.id1.fullName", "FirstName LastName");
lockPref("mail.identity.id1.smtpServer", "smtp1");
lockPref("mail.identity.id1.useremail", "username@domain.tld");
lockPref("mail.identity.id1.valid", true);


defaultPref("mail.account.account2.identities", "id1");
defaultPref("mail.accountmanager.accounts", "account1,account2");
```

I think with that you whould be able to reproduce.

**Comment by jobisoft** [comment]
What is autoadmin? Google returned different results but nothing looked official. Is this linux only?

**Comment by ohmer1** [comment]
Nop this is also working on Windows and OSX. Firefox share the same feature. See https://support.mozilla.org/en-US/kb/customizing-firefox-using-autoconfig

**Comment by jobisoft** [comment]
Before I start digging into this, could you make a diff of the profile folder to see what changes during shutdown, which than causes TbSync to fail? Could you manually undo these changes to make it work again?

**Comment by ohmer1** [comment]
The only changes I found that causes issue I found is the ~/.cache/thunderbird/<profile name>/startupCache directory.

**Comment by jobisoft** [comment]
I can reproduce it.

**Comment by jobisoft** [comment]
I released a new beta, where I think I fixed it. Can you confirm?
https://tbsync.jobisoft.de

**Comment by ohmer1** [comment]
Yes, the beta release is working great!

Thanks.

**Comment by jobisoft** [comment]
Released as TbSync 2.9
